mettle/src/stdapi/audio/apple_output.m

```diff
@@ -0,0 +1,64 @@
+#import <AVFoundation/AVFoundation.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <mettle.h>
+#include "channel.h"
+
+#include "output.h"
+
+AVAudioPlayer *audioPlayer;
+
+typedef struct context {
```

Comment:
The `new_audio_file` and `write_audio_file` methods and `context` struct are identical between `apple_output.m` and `linux_output.c`. Did you consider placing these implementations in a location that could be shared between the two files to avoid duplication?

Reply:
Very good point! Apologies this is my fault, I copied the apple_output.m code from linux_output.c :trollface:
Let's move the shared code to output.c
I can send a pull request tomorrow

Reply:
@timwr already fixed on my laptop :wink:, I made a `posix_output.c` file in the event we would support non posix systems

Reply:
That would be perfect. Feel free to push your fix :) 
Going forward we should find a way of streaming/buffering the data as it arrives instead of playing it when the channel closes.

Reply:
Sure, as soon as I can confirm it works as expected